src/verifiable-credential.js

```diff
@@ -39,24 +40,26 @@ class VerifiableCredential {
   }
 
   /**
-   * Add a context to this Credential's context array
+   * Add a context to this Credential's context array. Duplicates are omitted.
    * @param {string|object} context - Context to add to the credential context array
    * @returns {VerifiableCredential}
    */
   addContext(context) {
     ensureObjectWithKeyOrURI(context, '@context', 'context');
-    this.context.push(context);
+    this.context = getUniqueElementsFromArray([...this.context, context], JSON.stringify);
     return this;
   }
 
+
   /**
-   * Add a type to this Credential's type array
+   * Add a type to this Credential's type array. Duplicates are omitted.
    * @param {string} type - Type to add to the credential type array
    * @returns {VerifiableCredential}
    */
   addType(type) {
     ensureString(type);
-    this.type.push(type);
+    this.type = [...new Set([...this.type, type])];
```

Comment:
Similar comment as above. Rather then creating a Set and converting back to array each time an element is added, you can keep it as set and make it array in `toJSON`.

Reply:
Same as above.